server/src/main/java/org/elasticsearch/action/admin/cluster/allocation/ClusterAllocationExplanation.java

```diff
@@ -179,6 +185,9 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
                 }
                 builder.endObject();
             }
+            if (desiredNodeIds != null) {
+                builder.field("desired_node_ids", desiredNodeIds);
+            }
```

Comment:
This currently does the lazies thing by listing node ids like following:

```
{
  "index": "my-index",
  "shard": 0,
  "primary": true,
  "current_state": "started",
  "current_node": {
    "id": "node-id-0",
    "name": "node-name",
    "transport_address": "...",
    "attributes": {
      ...
    }
  },
  "desired_node_ids": ["node-id-0", "node-id-1"], <------------------------
  "can_remain_on_current_node": "no",
  "can_remain_decisions": [...],
  "can_move_to_other_node": "no",
  "move_explanation": "....",
  "node_allocation_decisions": [...]
}
```

Alternatively this could be node id and name pair or even objects of similar to `current_node` structure that include more verbose node description.

This information could be useful to answer following questions to understand if the shard is scheduled for relocation and what is the target (without having to look into output of any other endpoint).